ERCS/erc-7291.md

```diff
@@ -0,0 +1,696 @@
+- The **plugin components** implement additional functionality that only specific PBMs may require (e.g. logic to call external application programming interfaces to verify specific PBM condition was met, logic to track PBM usage patterns).
+- The **hook components** implement additional validation checks that some PBMs may need (e.g. checks for expiration, daily spending limit, goods & services limit etc).
+
+For example, a PBM creator may want to ensure only 50% of PBM Series A can be spent in supermarkets, while there are no restriction on proportion of PBM Series B that can be spent in supermarkets. He can implement a plugin smart contract to keep track of supermarket spendings by PBM users. He will also implement a hook to validate that less than 50% of the PBM Series A issued a user is spent in a supermarket. This allows him to use the same generic PBM wrapper and sovToken for both PBM Series A and B. In addition, PBM series A will register the plugin module and hook module for additional data tracking and validations.
+
+### Auditability
+
+PBM Wrapper **SHOULD** provide the public easily accessible mechanism(s) to verify the smart contract logic for unwrapping a PBM. Such mechanisms could be leveraged by automated validation or asynchronous user verifications from transacting parties and/or whitelisted third parties attestations.
+
+As the fulfilment of PBM conditions is likely to be subjected to audits to ensure trust amongst all transacting parties, the following evidence shall be documented to support audits:
+
+- The interface/events emitted **SHOULD** allow a fine-grained recreation of the transaction history, token types and token balances
+- The source code **SHOULD** be verified and formally published on a blockchain explorer.
+
+### Fungibility
+
+A PBM Wrapper **SHOULD** be able to wrap multiple types of compatible Spot Tokens (e.g. the same PBM Wrapper should be able to wrap USDC and XSGD). Spot Tokens wrapped by the same PBM wrapper may or may not be fungible to one another. The standard does NOT mandate how an implementation must do this.
+
+### PBM token details
+
+The ERC-1155 Multi Token Standard enables each token ID to correspond to a unique, configurable token type. All essential details facilitating the business or display logic for a specific PBM type **MUST** be defined for each token type. The mandatory fields for this purpose are outlined in the `struct PBMToken` (below). Future proposals may define additional, optional state variables as needed. Once a token detail has been defined, it **MUST** be immutable.
+
+Example of token details:
+
+```solidity
+pragma solidity ^0.8.0;
+
+abstract contract IPBMRC1_TokenManager {
```

Comment:
We generally prefer an interface over an abstract contract because the latter usually includes implementation details. Since this is an example, I think we can let it slide here.